lib/build-asset-map-reader.js

```diff
@@ -24,3 +24,25 @@ function readJSON(assetMap) {
 		});
 	});
 }
+
+// Retries a function that returns a promise
```

Comment:
I think if one cared one could lift this restriction by wrapping the call to `fn` in `Promise.resolve`.

Reply:
No wait, it would have to look like this:
```js
return Promise.resolve(...params).then(fn)
```

Anyway, this is probably more important for the promisified `setTimeout` because if its function argument throws we wouldn’t get a rejected promise.